rasa/core/actions/action.py

```diff
@@ -1142,6 +1142,9 @@ def _fails_unique_entity_mapping_check(
 
         form = FormAction(form_name, self._action_endpoint)
 
+        if slot_name not in form.required_slots(domain):
```

Comment:
@carlad I just thought if we should future-proof this by actually checking for the form's dynamically added slots too by using the `_add_dynamic_slots_requested_by_dynamic_forms` method in `FormAction`? I can transform this in a public method & add a docstring.

Reply:
Sounds good.

Reply:
Actually, after testing a few scenarios, this change wouldn't add value because dynamically added slots get requested straight away after they've been added by `FormValidationAction`, so the initial bug wouldn't apply to them. I'll keep it as it is 😄